### PR TITLE
Update modal

### PR DESCRIPTION
**Pull request by thedrew12**
- updated the modal to work better

### CODE REVIEW

File: src/components/Modal/modal.js [hunk 1]
```diff
@@ -10,85 +9,95 @@ import Icon from '../Icon';
...
-  onExit: Function,
   /** Display the content inside the modal */
   children: any,
   /** Click outsid the modal to close it, default true */
-  underlayClickExits?:boolean|void,
+  underlayClickExits?: boolean | void,
   /** Provide your main application node here (which the modal should render outside of) */
-  rootElementId?: string|void,
-  focusElementId?: string|void,
+  rootElementId?: string | void,
```
Comment (mhueter) [inline src/components/Modal/modal.js]: what's going on here? `|` is a bitwise-OR.

Comment (mhueter) [inline src/components/Modal/modal.js]: also the formatting of the ternaries is whack

File: src/components/Modal/modal.js [hunk 2]
```diff
@@ -10,85 +9,95 @@ import Icon from '../Icon';
...
+      isCard,
+      buttonClass
+    } = this.props;
 
     const extras = {
       focusDialog: !focusElementId
     };
 
     let underlayClass: string = styles['modal-underlay'];
+    let dialogContentClass: string = 'modal modal-animated';
```
Comment (mhueter) [inline src/components/Modal/modal.js]: is this TypeScript or something? What's up with the typed assignment?

Comment (thedrew12) [inline src/components/Modal/modal.js]: @mhueter It's Flow. https://flow.org/en/ We added it last year sometime.

File: src/components/Modal/modal.js (hunk 1 above)
Comment (thedrew12) [inline src/components/Modal/modal.js]: It is not a ternary, it is type checking the props. We are using this https://github.com/codemix/babel-plugin-typecheck, but it appears to be deprecated now.  :( Will need to address that. I liked it when we added it, but now I'm not a big fan of it. I think it just adds more work.

Comment (mhueter) [inline src/components/Modal/modal.js]: Alrighty, well we can address that later I suppose

**Comment by mhueter** [comment]
couple questions:

- What is the "activates with button and text in button and not a card" used for? Visually, it looks like a bug:
screen shot 2017-05-24 at 10 48 43 am

- I noticed the "X" is positioned outside the top right of the card, is that what JSON wanted?

**Comment by thedrew12** [comment]
@mhueter That modal story is used on the homepage where we don't want the youtube video to show up inside a Card type component with a white background. Most of the use cases will be used inside a Card. Yes, JSON was fine with the "X" outside the window. I did that partly because I wanted the "X" to be a single color, white instead of dynamically changing the color based on modal story you render.

**Pull request merged**